save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt

```diff
@@ -91,20 +95,30 @@ class DemoService(
     fun getAllDemos(): List<Demo> = demoRepository.findAll()
 
     /**
-     * @param demo
+     * @param demo [Demo] entity
+     * @param runCommands [RunCommandMap] where keys are demo mode names and values are run commands
      * @return [Demo] entity saved to database
      * @throws IllegalStateException if [demo] is already present in DB
      */
     @Transactional
-    fun saveIfNotPresent(demo: Demo): Demo = demoRepository.findByOrganizationNameAndProjectName(demo.organizationName, demo.projectName)?.let {
+    fun saveIfNotPresent(demo: Demo, runCommands: RunCommandMap): Demo = demoRepository.findByOrganizationNameAndProjectName(demo.organizationName, demo.projectName)?.let {
         throw IllegalStateException("Demo for project ${demo.organizationName}/${demo.projectName} is already added.")
-    } ?: save(demo)
+    } ?: run {
+        save(demo).also { savedDemo ->
+            runCommands.map { (mode, command) ->
+                RunCommand(savedDemo, mode, command)
+            }.let { runCommandList ->
+                runCommandRepository.saveAll(runCommandList)
```

Comment:
after saving `RunCommand`, do we need update `savedDemo` which `saveIfNotPresent` returns ?

Reply:
Why do we need to update it? We do not store `RunCommand` ids in `demo` table. AFAIK, each time we fetch `Demo` entity, a separate request is done to fetch `RunCommand` by `demo_id`

Reply:
`saveIfNotPresent` return `Demo` which is not valid -- it doesn't contains saved `RunCommand`
You didn't change `@PostMapping("/add")` and result of `saveIfNotPresent` is used as result of this endpoint

Reply:
Oh, got the point

Reply:
Done